src/pages/instances/InstanceOverviewDevices.tsx

```diff
@@ -0,0 +1,67 @@
+import { FC } from "react";
+import { MainTable } from "@canonical/react-components";
+import { LxdInstance } from "types/instance";
+import {
+  LxdGPUDevice,
+  LxdNoneDevice,
+  LxdOtherDevice,
+  LxdProxyDevice,
+} from "types/device";
+
+interface Props {
+  instance: LxdInstance;
+}
+type InstanceOverviewDevices =
+  | LxdNoneDevice
+  | LxdGPUDevice
+  | LxdOtherDevice
+  | LxdProxyDevice;
+
+const InstanceOverviewDevices: FC<Props> = ({ instance }) => {
+  const instanceDevices = Object.entries(
+    instance.expanded_devices ?? {},
+  ).filter(([, device]) => {
+    return !(device.type === "nic" || device.type === "disk");
+  });
```

Comment:
I think this should be in a helper function `getDevices(instance)` that returns an `LxdDeviceValue[]`. That way we don't need the cast as `const device = dev as InstanceOverviewDevices` below and contain the cast to a function.